addons/full-import/import.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+set -o nounset -o pipefail -o errexit
+
+source /usr/local/pf/addons/full-import/helpers.functions
+source /usr/local/pf/addons/full-import/database.functions
+source /usr/local/pf/addons/full-import/configuration.functions
+
+dump_path="$1"
+
+if ! [ -f "$dump_path" ]; then
+  echo "Please specify a valid archive to extract"
+  echo "Usage: import.sh /path/to/export.tgz"
+  exit 1
+fi
+
+extract_dir=`mktemp -d`
+
+cp -a $dump_path $extract_dir/export.tgz
+check_code $?
+
+pushd $extract_dir/
+check_code $?
+
+main_splitter
+
+echo "Extracting archive..."
+tar -xf export.tgz
+check_code $?
+
+echo "Found the following content in the archive:"
+ls -l
+
+main_splitter
+files_dump=`ls packetfence-files-*`
+echo "Found files dump '$files_dump'"
+
+echo "Extracting files dump"
+tar -xf $files_dump
+check_code $?
+
+main_splitter
+
+db_dump=`ls packetfence-db-dump-*`
+echo "Found compressed database dump '$db_dump'"
+gunzip $db_dump
+check_code $?
+db_dump=`ls packetfence-db-dump-*`
+echo "Uncompressed database dump '$db_dump'"
+
+main_splitter
+echo "Stopping PacketFence services"
+systemctl cat monit >/dev/null 2>&1 && (systemctl stop monit ; systemctl disable monit)
+systemctl isolate packetfence-base
+check_code $?
+systemctl stop packetfence-galera-autofix
+check_code $?
+
+if echo "$db_dump" | grep '\.sql$' >/dev/null; then
+  echo "The database dump uses mysqldump"
+  #TODO /tmp/grants.sql should be included in the export
+  import_mysqldump grants.sql $db_dump usr/local/pf/conf/pf.conf
+elif echo "$db_dump" | grep '\.xbstream$' >/dev/null; then
+  echo "The database uses mariabackup"
+  import_mariabackup $db_dump
+else
+  echo "Unable to detect format of the database dump"
+  exit 1
+fi
+
+#TODO: check the version of the export, we want to support only 10.3.0 and above
+#TODO: check if galera is enabled and stop if its the case
+
+main_splitter
+db_name=`get_db_name usr/local/pf/conf/pf.conf`
+upgrade_database $db_name
+check_code $?
+
+main_splitter
+restore_config_files `pwd`
+check_code $?
+
+main_splitter
+handle_network_change
+check_code $?
+
+main_splitter
+upgrade_configuration
+check_code $?
+
+main_splitter
+restore_profile_templates
```

Comment:
Is it on purpose to restore profile templates after upgrade_configuration ?

In the past, a script like `./addons/upgrade/to-8.3-conf-latin1-to-utf8.sh` modify these files.

Reply:
Good point, lets move it